cloudwatch_logger/src/log_node_param_helper.cpp

```diff
@@ -0,0 +1,199 @@
+      AWS_LOGSTREAM_WARN(
+      __func__,
+      "Whether to subscribe to rosout_agg topic configuration not found, setting to default value: "
+        << kNodeSubscribeToRosoutDefaultValue);
+      break;
+    case AwsError::AWS_ERR_OK:
+      AWS_LOGSTREAM_INFO(
+      __func__, "Whether to subscribe to rosout_agg topic is set to: " << subscribe_to_rosout);
+      break;
+    default:
+      subscribe_to_rosout = kNodeSubscribeToRosoutDefaultValue;
+      AWS_LOGSTREAM_ERROR(
+        __func__,
+        "Error " << ret 
+        << "retrieving parameter for Whether to subscribe to rosout_agg topic configuration " 
+        << ", setting to default value: " << kNodeSubscribeToRosoutDefaultValue);
+  }
+  return ret;
+}
+
+Aws::AwsError ReadMinLogVerbosity(
+  std::shared_ptr<Aws::Client::ParameterReaderInterface> parameter_reader,
+  int8_t & min_log_verbosity)
+{
+  min_log_verbosity = kNodeMinLogVerbosityDefaultValue;
+
+  std::string specified_verbosity;
+  Aws::AwsError ret =
+    parameter_reader->ReadStdString(kNodeParamMinLogVerbosityKey, specified_verbosity);
+  switch (ret)
+  {
+    case AwsError::AWS_ERR_NOT_FOUND:
+      AWS_LOGSTREAM_WARN(__func__, "Log verbosity configuration not found, setting to default value: "
+                                   << kNodeMinLogVerbosityDefaultValue);
+      break;
+    case AwsError::AWS_ERR_OK:
+      if ("DEBUG" == specified_verbosity) {
+        min_log_verbosity = rosgraph_msgs::Log::DEBUG;
+        AWS_LOG_INFO(__func__, "Log verbosity is set to DEBUG.");
+      } else if ("INFO" == specified_verbosity) {
+        min_log_verbosity = rosgraph_msgs::Log::INFO;
+        AWS_LOG_INFO(__func__, "Log verbosity is set to INFO.");
+      } else if ("WARN" == specified_verbosity) {
+        min_log_verbosity = rosgraph_msgs::Log::WARN;
+        AWS_LOG_INFO(__func__, "Log verbosity is set to WARN.");
+      } else if ("ERROR" == specified_verbosity) {
+        min_log_verbosity = rosgraph_msgs::Log::ERROR;
+        AWS_LOG_INFO(__func__, "Log verbosity is set to ERROR.");
+      } else if ("FATAL" == specified_verbosity) {
+        min_log_verbosity = rosgraph_msgs::Log::FATAL;
+        AWS_LOG_INFO(__func__, "Log verbosity is set to FATAL.");
+      } else {
+        ret = AwsError::AWS_ERR_PARAM;
+        AWS_LOGSTREAM_WARN(__func__,
+                          "Log verbosity configuration not valid, setting to default value: "
+                           << kNodeMinLogVerbosityDefaultValue);
+      }
+      break;
+    default:
+      AWS_LOGSTREAM_ERROR(__func__, 
+        "Error " << ret << " retrieving log verbosity configuration, setting to default value: "
+        << kNodeMinLogVerbosityDefaultValue);
+  }
+  return ret;
+}
+
+Aws::AwsError ReadSubscriberList(
+  std::shared_ptr<Aws::Client::ParameterReaderInterface> parameter_reader,
+  std::vector<ros::Subscriber> & subscriptions, ros::NodeHandle & nh, bool & subscribe_to_rosout,
```

Comment:
subscribe_to_rosout should be const and marked placed in the front of the parameter list.

Reply:
I didn't notice that as I moved the code, I changed that to a `bool` by value as that is a primitive type

```c++
Aws::AwsError ReadSubscriberList(
  bool subscribe_to_rosout,
  std::shared_ptr<Aws::Client::ParameterReaderInterface> parameter_reader,
  boost::function<void(const rosgraph_msgs::Log::ConstPtr &)> callback,
  ros::NodeHandle & nh,
  std::vector<ros::Subscriber> & subscriptions);
```

is that what you mean?

Reply:
sure, but I would still mark subscribe_to_rosout const in the cpp file to show readers that the parameter does not get modified in the function.